### PR TITLE
Fixed HTTP server error and auto increment port in command line 256 & 239.

### PR DESCRIPTION
**Pull request by vineethbc**
The http server error is being handled gracefully with error message & reload increments the port & tries again.
https://github.com/alallier/reload/issues/256
https://github.com/alallier/reload/issues/239

### CODE REVIEW

File: bin/reload [hunk 1]
```diff
@@ -30,9 +30,18 @@ if (typeof program.watchDir === 'undefined') {
...
 
-console.log('\nReload web server:')
-console.log('listening on port ' + clc.blue.bold(program.port))
-console.log('monitoring dir ' + clc.green.bold(program.dir))
+  console.log('\nReload web server:')
+  console.log('listening on port ' + clc.blue.bold(program.port))
+  console.log('monitoring dir ' + clc.green.bold(program.dir))
+
+  supervisor.child.on('exit', function () {
+    program.port++
```
Comment (alallier) [inline bin/reload:43]: Can we add an on by default flag option for the auto increment?

**Review by alallier — CHANGES_REQUESTED** [review]
This all looks good, just one feedback/action item

**Comment by porst17** [comment]
Any chance for this being merged soon?

I created something like this in https://github.com/porst17/reload/commit/549e4fbb9682b7351e944946e4f612d09bb52e8d using the `portfinder` package, not being aware of the work done in this pull request.

But having a solution like yours that doesn't add another dependency is probably preferable.

**Comment by alallier** [comment]
@porst17 I have this basically done and re-worked in a separate branch. I should be able to get this up soon.

**Pull request closed without merging**